### PR TITLE
Clarify CLI help-text to specify limited role selection

### PR DESCRIPTION
**Pull request by arunsundark**
Test results:
~~~sonic# configure terminal
sonic(config)# username testadmin password sonic role admin
sonic(config)# username testoper1 password sonic role operator
sonic(config)# username testowner password sonic role owner
~~~
~~~
admin@sonic:~$ cat /etc/passwd | grep test
testadmin:x:1005:1000::/home/testadmin:/usr/bin/sonic-launch-shell
testoper1:x:1006:37::/home/testoper1:/usr/bin/sonic-launch-shell
testowner:x:1007:37::/home/testowner:/usr/bin/sonic-launch-shell
~~~

### CODE REVIEW

**Comment by jeffyin-dell** [comment]
Can we also fix the help-string in the CLI to specify at least the currently-supported roles?

**Comment by arunsundark** [comment]
Sure

File: src/translib/transformer/host_acct_mgr.go [hunk 1]
```diff
@@ -15,12 +15,8 @@ func roleToGroup(role string) []string {
 	switch role {
 	case "admin":
 		return []string{"admin", "sudo", "docker"}
-	
-	case "operator":
-		return []string{"operator", "docker"}
-	
 	default:
-		return []string{}
+		return []string{"operator", "docker"}
```
Comment (jeffyin-dell) [inline src/translib/transformer/host_acct_mgr.go:19]: Don't want to assign users by default to the `operator` group. Rather, they should be assigned to the same group as the provided role (plus `docker`).

i.e., if I configure `username jeffyin password <password> role xyz123`, I should, as a result, be assigned to the `xyz123` and `docker` groups. The system (currently) would then treat me as a non-admin user.

**Review by jeffyin-dell — CHANGES_REQUESTED** [review]

File: src/translib/transformer/host_acct_mgr.go (hunk 1 above)
Comment (arunsundark) [reply src/translib/transformer/host_acct_mgr.go:19]: Hamd does not accept any role other than admin and operator.

**Review by jeffyin-dell — APPROVED** [review]

**Review by nirenjan — APPROVED** [review]

File: src/CLI/clitree/cli-xml/system.xml [hunk 2]
```diff
@@ -35,7 +35,7 @@
     <COMMAND name="username" help="Add new user" ptype="SUBCOMMAND" mode="subcommand">
        <PARAM name="user-name" help="User name string" ptype="STRING"></PARAM>
            <PARAM name="password" help="Clear text password " ptype="SUBCOMMAND" mode="subcommand"> <PARAM name="passwd" help="Type clear text password string" ptype="STRING"></PARAM>
-               <PARAM name="role" help="User role" ptype="SUBCOMMAND" mode="subcommand"> <PARAM name="rl" help="Type User role string" ptype="STRING">
+               <PARAM name="role" help="User role" ptype="SUBCOMMAND" mode="subcommand"> <PARAM name="rl" help="Type User role string, allowed roles are admin and operator" ptype="STRING">
```
Comment (jeffyin-dell) [inline src/CLI/clitree/cli-xml/system.xml]: Can you revise the helptext to `User role (admin/operator)`

We don't need to tell the user to type a string in the help-text :)

**Review by jeffyin-dell — CHANGES_REQUESTED** [review]

**Review by jeffyin-dell — APPROVED** [review]

**Pull request merged**